src/accelerate/utils/operations.py

```diff
@@ -169,19 +167,28 @@ def send_to_device(tensor, device, non_blocking=False, skip_keys=None):
                 for k, t in tensor.items()
             }
         )
-    elif hasattr(tensor, "to"):
-        if is_npu_available():
+    elif is_torch_tensor(tensor) or hasattr(tensor, "to"):
+        # `torch.Tensor.to("npu")` could not find context when called for the first time (see this [issue](https://gitee.com/ascend/pytorch/issues/I8KECW?from=project-issue)).
+        if device == "npu":
+            device = "npu:0"
+        if device == "xpu":
+            device = "xpu:0"
+        try:
+            return tensor.to(device, non_blocking=non_blocking)
+        except TypeError:  # .to() doesn't accept non_blocking as kwarg
+            return tensor.to(device)
+        except AssertionError:
```

Comment:
Here, we implicitly make the assumption that `AssertionError` is always associated with NPU and all other errors with XPU, right? Otherwise, it looks like strange things would happen. Hopefully this is a robust assumption.

Reply:
Unfortunately yes - when I sent the first version of this PR only the NPU check existed. I don't know where the XPU code lives that raises this error and I don't have a device to check, would you have a pointer?